### PR TITLE
Fix/3218/doctype

### LINKED ISSUE #3218: [Severity Updates] hint-doctype
| Report | Severity |
| --------| ----------|
| Report message when a doctype is duplicated: 'doctype' is not needed as one was already specified. | Warning |
| Report message when a doctype is not specified: 'doctype' was not specified. | Error |
| Report message when a doctype is not specified as '<!doctype html>': 'doctype' should be specified as '<!doctype html>'. | Error if resulting docmode is not standards, warning otherwise |
| Report message when a doctype is not the first thing in the html: 'doctype' should be specified before anything else. | Error if resulting docmode is not standards, warning otherwise | 
| Report message when the resource has no content: Resource has no content. | Off |

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- [x] Added/Updated related tests.

## Short description of the change(s)

Fix #3218

Needs to be rebased and not squashed.

### CODE REVIEW

File: packages/utils-dom/tests/html.ts [hunk 1]
```diff
@@ -28,6 +28,11 @@ test('HTMLDocument.base should return the base url', (t) => {
     t.is(t.context.document.base, 'http://example.com/resources/');
 });
 
+test('HTMLDocument.compatMode should return the compat mode', (t) => {
+    t.is(t.context.document.compatMode, 'CSS1Compat');
```
Comment (antross) [inline packages/utils-dom/tests/html.ts]: Let's add a test case for a `BackCompat` scenario too.

Comment (molant) [reply packages/utils-dom/tests/html.ts]: done, I forced push so we could rebase directly from the website once CI passes.

**Comment by molant** [comment]
@sarvaje why a breaking change? We are adding new features 🤔

**Comment by sarvaje** [comment]
Now you will need to add the html parser to your configuration, isn't it?

**Comment by molant** [comment]
I should, not sure why the linter is not picking up the missing dependency 🤔

**Pull request merged**